src/core/load-fragment.c

```diff
@@ -283,6 +283,23 @@ int config_parse_unit_path_strv_printf(
         assert(rvalue);
         assert(u);
 
+        if (isempty(rvalue)) {
+                char **empty;
+
+                /* Empty assignment resets the list. As a special rule
+                 * we actually fill in a real empty array here rather
+                 * than NULL, since some code wants to know if
+                 * something was set at all... */
```

Comment:
I am pretty sure that when the user assigns the empty string we should really reset things to the state when really never anything was set at all. With other words, I think we actually *should* just assign NULL here, so that assigning the empty string really fully undoes anything set before

Reply:
I just copied from `config_parse_strv()` in conf-parser.c.
https://github.com/systemd/systemd/blob/6297d07b82baca4b5602076869477e0de7e1443b/src/shared/conf-parser.c#L771-L786
Do you think we should not set an empty array at `config_parse_strv()`?

Reply:
Anyway, I agree with you. I will update it.

Reply:
Similary, `config_parse_user_group_strv()` also set an empty array.
https://github.com/systemd/systemd/blob/c792ec2e3512a672881fc847ff432e26b641c9c9/src/core/load-fragment.c#L1953-L1964